bundles/EcommerceFrameworkBundle/VoucherService/TokenManager/Pattern.php

```diff
@@ -541,10 +541,26 @@ public function prepareConfigurationView(&$viewParamsBag, $params)
 
         /** @var PaginatorInterface $paginator */
         $paginator = \Pimcore::getContainer()->get(\Knp\Component\Pager\PaginatorInterface::class);
+
+        $page = $params['page'] ?? 1;
+        $perPage = isset($params['tokensPerPage']) ? (int)$params['tokensPerPage'] : 25;
+
+        $total = count($tokens);
+
+        $availablePages = ceil($total / $perPage);
+
+        if ($page > $availablePages) {
+            $page = $availablePages;
+        }
+
+        if ((int) $availablePages === 0) {
+            $page = 1;
+        }
```

Comment:
```suggestion
        $page = isset($params['page']) ? (int)$params['page'] : 1;
        $perPage = isset($params['tokensPerPage']) ? (int)$params['tokensPerPage'] : 25;

        $total = count($tokens);

        $availablePages = (int) ceil($total / $perPage);

        if ($page > $availablePages) {
            $page = $availablePages;
        }

       $page = $page ?: 1;
```

- added isset and cast to `params['page']`, otherwise would appear as a string 
- convert the ceil result of `availablePages` to int, to remove the cast below (for `===0` check) and because `$page` should be int (and would be float after `$page = $availablePages`).
- added elvis check for `$page`, as it should be never 0 but starts from 1, therefore not limiting this check to only when availablePages is 0

See also https://github.com/KnpLabs/knp-components/blob/6b6efa81ee894e325744bf785d50dc962937b1f2/src/Knp/Component/Pager/PaginatorInterface.php#L39-L40